disdat/apply.py

```diff
@@ -421,7 +450,9 @@ def main(args):
         print("Apply unavailable -- Disdat not in a valid context.")
         return
 
-    dynamic_params = json.dumps(common.parse_params(args.params))
+    # Create a dictionary of str->str arguments
+    user_params = common.parse_params(args.params) # a dictionary of str:str
+    deser_user_params = convert_str_params(args.pipe_cls, user_params)
```

Comment:
i'm not sure why you separate this into two steps - you never need (want) the str params return from common.parse_params - you would always convert them immediately.  this forces the developer to know to call both functions in sequence.  i would combine their functionality to create a cleaner interface:  take the command line arguments and return the deserialized parameters

Reply:
Meh, I think it makes more sense as two separate functions.   Otherwise convert_str_params takes in some janky params from argparse.